### DISCUSSION TITLE
same color scheme for markers and legend when using the style_function

### DISCUSSION THREAD

**Discussion by kbzsl** [geopandas/geopandas#2759] [Q&A] [answered]
I am using the style_function from GeoDataFrame.explore() to set the colors of point/markers based on a categorical variable from the dataframe.  

`'style_function':lambda x: {'color':color_dict.get(x['properties']['category'])}`

The point/marker coloring is working ok, but the colors in the legend are totally different (it looks like some default colors).

How one can "synchronize" the colors of the markers and the colors from the legend? In another words, how can the same colors based on a style_function applied to the legend?

**Comment by martinfleis** [comment, marked as answer]
In this case, geopandas has no awareness on what you're mapping onto points, so it cannot create the legend for you. You will need to do that manually. You can probably reuse the function we use internally here https://github.com/geopandas/geopandas/blob/7e1164fff8d85ef48f857c10a648188b3692b56e/geopandas/explore.py#L752

There was a discussion on making it public anyway, so you should be able import and use it without much trouble.

**Reply by kbzsl** [reply]
Thank you, I will look into that recommended function.

In meantime, I found a workaround, and I hope that it generalizes well (at least it works for my current application).

It looks to me that the items on the legend are (always?) in an alphabetical order. Therefore I created a ordered list of distinct elements of that categorical variable, and I used that list to create a list of colors for the `cmap `parameter.
```
color_dict = {'catA':'blue', 'catB':'orange', 'catC':'red'}

color_map = list(map(color_dict.get, sorted(list(gdf_table['category'].unique()))))

gdf_table.explore(
	cmap=colors.ListedColormap(color_map), 
	...
	)
```
